src/dat.c

```diff
@@ -193,7 +199,13 @@ int eraDat(int iy, int im, int id, double fd, double *deltat )
    };
 
 /* Number of Delta(AT) changes */
-   enum { NDAT = (int) (sizeof changes / sizeof changes[0]) };
+   enum { _NDAT = (int) (sizeof _changes / sizeof _changes[0]) };
+   
+/* Initialise leap seconds if needed */
+   if ( NDAT == 0) {
+      NDAT = _NDAT;
+      changes = (eraLEAPSECOND *)_changes;
+   }
```

Comment:
The first time `eraDat` is called, and if eraUpdateLeapSeconds() has not been called before, `NDAT` and `changes` are made to point to the private and static `_NDAT` and `_changes`. This preserves the current behavior of SOFA/ERFA.

Reply:
This is where one may want to had a hook to read and parse a distribution provided leap second file, like `zoneinfo/leap-seconds.list` on Debian. If `NDAT==0`, we are on the first call, and before setting `NDAT` to `_NDAT` and `changes` to `_changes`, one could try to get the update from such files.